content/blog/2018-05-02-onboarding-is-work.md

```diff
@@ -0,0 +1,325 @@
+```
+
+We have self-reported reviewing times for 136 of 184 reviews.
+
+``` r
+ggplot(airtable) +
+  geom_boxplot(aes(y = review_hours, x = "")) +
+  hrbrthemes::theme_ipsum(base_size = 16,
+                          axis_title_size = 16) +
+  theme(axis.title.x=element_blank()) +
+  ylab("Reviewing time (hours)")
+```
+
+![hours spent
+reviewing](/img/blog-images/2018-05-02-onboarding-is-work/unnamed-chunk-6-1.png)
+
+As we found before, it appears to take our reviewers a similar amount of
+time to review R packages as [scientists take to review a manuscript (5
+hour median and 9 hour
+mean)](http://publishingresearchconsortium.com/index.php/112-prc-projects/research-reports/peer-review-in-scholarly-journals-research-report/142-peer-review-in-scholarly-journals-perspective-of-the-scholarly-community-an-international-study).
+
+One potential question is whether reviewer time is affected by the size
+of the package reviewed as measured by for instance number of exports
+(classes and functions).
+
+``` r
+get_namespace <- function(package_name){
+  message(package_name)
+  local_path_archive <- paste0(getwd(), "/repos_at_submission/", package_name)
+  if(length(fs::dir_ls(local_path_archive)) != 0){
+    ns <- devtools::parse_ns_file(local_path_archive)
+    tibble::tibble(package = package_name, 
+                   exports = length(ns$exports) +
+                     length(ns$exportClasses))
+  }else{
+    return(NULL)
+  }
+}
+
+packages <- fs::dir_ls("repos_at_submission") 
+packages <- stringr::str_replace_all(packages, "repos_at_submission\\/", "")
+purrr::map_df(packages, get_namespace) %>%
+  readr::write_csv("output/namespace.csv")
+```
+
+Interestingly, we find no relationship between the reviewing time and
+number of exports:
+
+``` r
+namespace_ro <- readr::read_csv("output/namespace.csv")
+namespace_ro <- dplyr::left_join(namespace_ro, airtable, by = "package")
+ggplot(namespace_ro) +
+  geom_point(aes(exports, review_hours)) +
+  hrbrthemes::theme_ipsum(base_size = 16,
+                          axis_title_size = 16) +
+  ylab("Reviewing time (hours)") +
+  xlab("No. of exports")
+```
+
+![](/img/blog-images/2018-05-02-onboarding-is-work/scatterplot-size-vs-reviewing-time-1.png)
+
+There are a few potential explanations to this that might be fruitful to
+explore. For instance, does this mean that there’s only so much time,
+and so larger packages get less scrutiny per line of code? Or does
+review time just depend more on the reviewer than the package?
+
+### Work done by editors
+
+Editors manage the review process, performing initial package checks,
+identifying and contacting reviewers, and then moderating the cajoling
+the process forward. Our best measure for editor effort is the number of
+packages handled by an editor in a given time frame, which we can track
+from the assignments on GitHub issues since the start of onboarding.
+Looking at this over time, we can see how editor workloads changed in
```

Comment:
> how editor workloads changed in response to growing number of assignments

looking at this plot, I immediately wondered about the total number of assignments in each of those intervals to indicate how much the overall throughput and effort has increased. Is it not tooooo difficult to add that to the plot? Or in the text